buildSrc/src/main/java/org/elasticsearch/gradle/VersionCollection.java

```diff
@@ -254,6 +255,17 @@ private String getBranchFor(Version version) {
             }
         }
 
+        // Special case when  we are about to release a new major, we have branched and incremented versions
```

Comment:
As @rjernst mentioned, what do we actually use these versions for? They are never included in any of the BWC tests. Basically, if an unreleased version isn't wire or index compatible it's kind of irrelevant isn't it?